### PR TITLE
CN0548 demo

### PR DESCRIPTION
**Pull request by amiclaus**
EVAL-ADICUP3029 demo project:

- no-OS submodule update with latest AD7799 driver changes
- Main application
- Makefiles for build integration
- Python script with GUI allowing the user to interface with the device via IIO

Signed-off-by: Antoniu Miclaus <antoniu.miclaus@analog.com>

### CODE REVIEW

File: projects/ADuCM3029_demo_cn0548/src/cn0548.c [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+#include "error.h"
+#include "iio.h"
+#include "iio_app.h"
+#include "irq.h"
+#include "irq_extra.h"
+#include "uart.h"
+#include "uart_extra.h"
+#include "platform_init.h"
+#include "timer.h"
+#include "util.h"
```
Comment (mchindri) [inline projects/ADuCM3029_demo_cn0548/src/cn0548.c:59]: You only need: 
spi
platform_init
ad7799
iio_add799
iio_app
error.h
The other includes can be removed.

File: projects/ADuCM3029_demo_cn0548/builds.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    "cn0548": ""
```
Comment (mchindri) [inline projects/ADuCM3029_demo_cn0548/builds.json:2]: nitpick: I would use iio or demo as name and then the final .hex name will be: ADuCM3029_demo_cn0548_iio.
not ADuCM3029_demo_cn0548_cn0548

**Review by mchindri — APPROVED** [review]
Just 2 small issues

File: projects/ADuCM3029_demo_cn0548/builds.json (hunk 2 above)
Comment (amiclaus) [reply projects/ADuCM3029_demo_cn0548/builds.json:2]: cool, will do.

File: projects/ADuCM3029_demo_cn0548/src/cn0548.c (hunk 1 above)
Comment (amiclaus) [reply projects/ADuCM3029_demo_cn0548/src/cn0548.c:59]: Sure.

**Comment by amiclaus** [comment]
V2:
- remove unnecessary includes
- change `builds.json` string

**Comment by amiclaus** [comment]
V3:
- minor fix addressing the python script

**Review by amiclaus — COMMENTED** [review]
Remark: 
- the python script uses pyadi-iio module. A PR was added for ad7799 support here:
https://github.com/analogdevicesinc/pyadi-iio/pull/186

**Pull request merged**